vaadin-date-picker-flow-parent/vaadin-date-picker-flow-integration-tests/src/main/java/com/vaadin/flow/component/datepicker/DatePickerFallbackParserPage.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2000-2024 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.datepicker;
+
+import java.time.LocalDate;
+
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.data.binder.Result;
+import com.vaadin.flow.router.Route;
+
+import elemental.json.Json;
+import elemental.json.JsonObject;
+
+@Route("vaadin-date-picker/fallback-parser")
+public class DatePickerFallbackParserPage extends Div {
+    public DatePickerFallbackParserPage() {
+        DatePicker datePicker = new DatePicker();
+        datePicker.setFallbackParser((s) -> {
+            if (s.equals("newyear")) {
```

Comment:
This seems to be a limitation in how DatePicker is currently designed. It assumes that date parsing is done synchronously on the client-side. To highlight the date when entering something like `newyear`, we would need to send a server request on every keystroke, which could lead us to moving the entire parsing logic to the server.

Reply:
My comment was mostly just to point out the obvious, I know there is no easy solution for it. This is something that a developer or user might notice and not understand what the difference is compared to entering some other format. Especially considering that someone could also use this feature to parse a custom date format such as `14.11.2024` instead of using I18N date formats.

Anyway, I don't think it's a blocker for this PR. As discussed we can try to migitate this by not showing a highlight if the date can not be parsed by the client, or at some later point research other solutions if this turns out to be a problem.